### PR TITLE
[JENKINS-72028] at ` test/src/test/java/hudson/` replace deprecated `URL()`  

### PR DESCRIPTION
**Pull request by abhishekmaity**
Replace deprecated `URL()` with `URI`#`.toURL()` . Changes implemented for [test/src/test/java/hudson/‎](https://github.com/search?q=repo%3Ajenkinsci%2Fjenkins+%22new+URL%22+lang%3Ajava+path%3A%2F%5Etest%5C%2Fsrc%5C%2Ftest%5C%2Fjava%5C%2F%2F+path%3A%2F%5Etest%5C%2Fsrc%5C%2Ftest%5C%2Fjava%5C%2Fhudson%5C%2F%2F&type=code&p=1) files only.

See [JENKINS-72028](https://issues.jenkins.io/browse/JENKINS-72028).

### Proposed changelog entries

- JENKINS-X72028, replaced deprecated `URL()` with `URI#.toURL()`

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by abhishekmaity** [comment]
requesting @jtnord @timja @NotMyFault

**Comment by abhishekmaity** [comment]
@jtnord I have applied the highlighted changes

File: test/src/test/java/hudson/model/UpdateSiteTest.java
Comment (jtnord) [inline test/src/test/java/hudson/model/UpdateSiteTest.java:134]:
```
2023-10-06T15:36:43.954Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.3.0:check (validate) on project jenkins-test: Failed during checkstyle configuration: Exception was thrown while processing /home/jenkins/agent/workspace/Core_jenkins_PR-8569/test/src/test/java/hudson/model/UpdateSiteTest.java: IllegalStateException occurred while parsing file /home/jenkins/agent/workspace/Core_jenkins_PR-8569/test/src/test/java/hudson/model/UpdateSiteTest.java. 117:27: mismatched input '(' expecting ';': InputMismatchException -> [Help 1]
```

missing quote?
```suggestion
        baseUrl = new URI("http", null, "localhost", connector.getLocalPort(), RELATIVE_BASE, null, null).toURL();
```

**Review by jtnord — APPROVED** [review]

**Review by NotMyFault — COMMENTED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**
